evennia/contrib/clothing.py

```diff
@@ -0,0 +1,678 @@
+"""
+Clothing - Provides a typeclass and commands for wearable clothing,
+which is appended to a character's description when worn.
+
+Evennia contribution - Tim Ashley Jenkins 2017
+
+Clothing items, when worn, are added to the character's description
+in a list. For example, if wearing the following clothing items:
+
+    a thin and delicate necklace
+    a pair of regular ol' shoes 
+    one nice hat
+    a very pretty dress
+
+A character's description may look like this:
+
+    Superuser(#1)
+    This is User #1.
+    
+    Superuser is wearing one nice hat, a thin and delicate necklace, 
+    a very pretty dress and a pair of regular ol' shoes.
+    
+Characters can also specify the style of wear for their clothing - I.E.
+to wear a scarf 'tied into a tight knot around the neck' or 'draped
+loosely across the shoulders' - to add an easy avenue of customization.
+For example, after entering:
+
+    wear scarf draped loosely across the shoulders
+    
+The garment appears like so in the description:
+
+    Superuser(#1)
+    This is User #1.
+
+    Superuser is wearing a fanciful-looking scarf draped loosely
+    across the shoulders.
+
+Items of clothing can be used to cover other items, and many options
+are provided to define your own clothing types and their limits and
+behaviors. For example, to have undergarments automatically covered
+by outerwear, or to put a limit on the number of each type of item
+that can be worn. The system as-is is fairly freeform - you
+can cover any garment with almost any other, for example - but it
+can easily be made more restrictive, and can even be tied into a
+system for armor or other equipment.
+
+To install, import this module and have your default character
+inherit from ClothedCharacter in your game's characters.py file:
+
+    from evennia.contrib.clothing import ClothedCharacter
+
+    class Character(ClothedCharacter):
+        
+And do the same with the ClothedCharacterCmdSet in your game's
+default_cmdsets.py:
+
+    from evennia.contrib.clothing import ClothedCharacterCmdSet
+
+    class CharacterCmdSet(default_cmds.CharacterCmdSet):
+        
+From here, you can use the default builder commands to create clothes
+with which to test the system:
+    
+    @create a pretty dress : evennia.contrib.clothing.Clothing
+    @set dress/clothing_type = 'body'
```

Comment:
I'd say "clothing type" is better, as this system allows you to specify where a garment is worn - say, perhaps, a scarf worn around the waist, around the head, or around the neck - but I think the confusion comes from 'body' being the name of the clothing types for dresses and bodysuits is pretty misleading. I'll either rename that slot or give a different example.